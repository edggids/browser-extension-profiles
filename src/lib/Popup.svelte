<script lang="ts">
    export let isOpen = false;
    let className = '';

    export { className as class};
</script>

<div role="dialog" class={`${className} ${isOpen ? '': 'hidden'} p-2 bg-ground rounded shadow z-10`}>
    <slot></slot>
</div>

