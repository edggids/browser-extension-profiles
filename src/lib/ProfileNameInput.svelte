<script lang="ts">
    let className = '';
    export { className as class };

    export let editable = false;
    export let value = '';
</script>

<div data-testid="inputGroup" class={`container transition-bg bg-current-light transition-color text-current text-lg pt-2 pb-3.5 ${className}`}>
    <span class="text-xl">{'{'}</span>
    {#if editable}
        <input placeholder="SET NAME HERE" class="plain" type="text" bind:value />
    {:else}
        <span>{ value }</span>
    {/if}
    <span class="text-xl">{'}'}</span>
</div>
