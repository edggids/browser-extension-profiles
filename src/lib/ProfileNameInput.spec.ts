import {cleanup, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach} from 'vitest';
import ProfileNameInput from "./ProfileNameInput.svelte";
import Popup from "./Popup.svelte";

describe('ProfileNameInput component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container } = render(ProfileNameInput);

    expect(container).toMatchSnapshot();
  });

  it('renders as non-editable when editable is false', () => {
    const { container } = render(ProfileNameInput, { props: { editable: false } });
    const input = container.querySelector('input');
    expect(input).toBeNull();
  });

  it('renders as editable when editable is true', () => {
    const { container } = render(ProfileNameInput, { props: { editable: true } });
    const input = container.querySelector('input');
    expect(input).not.toBeNull();
  });

  it('renders with provided value when editing', () => {
    const { container } = render(ProfileNameInput, { props: { editable: true, value: 'Test Value' } });
    const input = container.querySelector('input');
    expect(input.value).toBe('Test Value');
  });

  it('renders with provided value when not editing', () => {
    const { container, getByText } = render(ProfileNameInput, { props: { editable: false, value: 'Test Value' } });
    expect(getByText('Test Value')).not.toBeNull();
  });

  it('applies custom class when provided', () => {
    const { container, getByTestId } = render(ProfileNameInput, { props: { class: 'custom-class' } });
    const wrapper = getByTestId('inputGroup');

    expect(wrapper.classList.contains('custom-class')).toBe(true);
  });
});
