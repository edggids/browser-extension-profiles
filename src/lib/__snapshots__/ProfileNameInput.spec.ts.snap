// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileNameInput component > mounts 1`] = `
<body>
  <div>
    <div
      class="container transition-bg bg-current-light transition-color text-current text-lg pt-2 pb-3.5 "
      data-testid="inputGroup"
    >
      <span
        class="text-xl"
      >
        {
      </span>
       
      <span>
        
      </span>
       
      <span
        class="text-xl"
      >
        }
      </span>
    </div>
  </div>
</body>
`;
